src/common.jl

```diff
@@ -186,40 +186,28 @@ function DiffEqBase.solve(
         warned && warn_compat()
     end
 
-    sizeu = size(prob.u0)
-    # This if block handles DynamicalODEProblems
-    # credit to @SebastianM-C for coming up with this solution
-    if prob.f isa DynamicalODEFunction && !isinplace
-        _alg = _TaylorMethod(alg.order, parse_eqs = false)
-        f = (du,u,p,t) ->
-        @inbounds begin
-            dv1 = view(du, firstindex(du):lastindex(du)÷2)
-            dv2 = view(du, lastindex(du)÷2+1:lastindex(du))
-            v1 = view(u, firstindex(u):lastindex(u)÷2)
-            v2 = view(u, lastindex(u)÷2+1:lastindex(u))
-            dv1 .= prob.f.f1(v1, v2, p, t)
-            dv2 .= prob.f.f2(v1, v2, p, t)
-            return nothing
-        end
-        u0 = convert(Array{eltype(prob.u0)}, prob.u0)
-        _prob = ODEProblem(f, u0, prob.tspan, prob.p; prob.kwargs...)
-        # DiffEqBase.solve(prob, _alg, args...; kwargs...)
-        integrator = DiffEqBase.__init(_prob, _alg, args...; kwargs...)
-    else
-        f = prob.f
-        if !isinplace && typeof(prob.u0) <: AbstractArray
-            f! = (du, u, p, t) -> (du .= f(u, p, t); 0)
+    f = prob.f
+    if !isinplace && typeof(prob.u0) <: AbstractArray
```

Comment:
Is `typeof(prob.u0) <: AbstractArray` necessary?
I think that the only case that we have `prob.u0` not be an array is when it is a scalar differential equation, but in that case it can only be out-of-place, so `!isinplace` is already true.

Reply:
Thank you for your suggestion. If we think an oop scalar problem, and if we removed that `typeof(prob.u0) <: AbstractArray`, then in that case the code would try to transform it into a in-place scalar problem, but currently TaylorIntegration does not support in-place scalar problems; that's why I left it there. Tests actually break if I remove that part